mne/preprocessing/tests/test_csd.py

```diff
@@ -0,0 +1,158 @@
+# PyCSD
+#
+# Copyright 2003-2010 Jürgen Kayser <rjk23@columbia.edu>
+# Copyright 2017 Federico Raimondo <federaimondo@gmail.com> and
+#                Denis A. Engemann <dengemann@gmail.com>
+#
+# The following code is a derivative work of the code from the CSD Toolbox,
+# which is licensed GPLv3. This code therefore is also licensed under the terms
+# of the GNU Public License, version 3.
+#
+# The original CSD Toolbox can be find at
+# http://psychophysiology.cpmc.columbia.edu/Software/CSDtoolbox/
+
+import os.path as op
+
+import numpy as np
+from scipy import io as sio
+
+import pytest
+from numpy.testing import assert_allclose, assert_equal
+import mne
+from mne.channels import make_standard_montage
+from mne import create_info
+from mne.io import RawArray
+from mne.utils import run_tests_if_main
+from mne.datasets import testing
+
+from mne.preprocessing import compute_current_source_density
+
+from mne.channels.interpolation import _calc_g, _calc_h
+from mne.preprocessing._csd import _compute_csd
+from mne.preprocessing._csd import _prepare_G
+
+base_path = op.join(testing.data_path(download=False), 'preprocessing')
+
+mat_contents = sio.loadmat(op.join(base_path, 'test-eeg.mat'))
+data = mat_contents['data']
+n_channels, n_epochs = data.shape[0], data.shape[1] // 386
+sfreq = 250.
+ch_names = ['E%i' % i for i in range(1, n_channels + 1, 1)] + ['STI 014']
+ch_types = ['eeg'] * n_channels + ['stim']
+data = np.r_[data, data[-1:]]
+data[-1].fill(0)
+info = create_info(ch_names=ch_names, ch_types=ch_types, sfreq=sfreq)
+raw = RawArray(data=data, info=info)
+montage = make_standard_montage('GSN-HydroCel-257')
+raw.set_montage(montage)
```

Comment:
This all needs to be inside the test, otherwise it executes at test collection time